src/ensembl/src/content/app/browser/browserActions.ts

```diff
@@ -395,3 +395,36 @@ export const toggleRegionEditorActive = createAction(
 export const toggleRegionFieldActive = createAction(
   'browser/toggle-region-field-active'
 )<boolean>();
+
+export const clearActiveGenomeId = createAction(
+  'browser/clear-active-genome-id'
+)();
+
+export const clearActiveEnsObjectIdsForGenome = createAction(
+  'browser/clear-active-ens-object-ids-for-genome'
+)<string>();
+
+export const deleteSpeciesAndSaveBrowser = (
+  genomeIdToRemove: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    if (activeGenomeId === genomeIdToRemove) {
+      dispatch(clearActiveGenomeId);
+      browserStorageService.clearActiveGenomeId();
+    }
+
+    const currentActiveEnsObjectIds = getBrowserActiveEnsObjectIds(state);
+    const updatedActiveEnsObjectIds = {
+      ...currentActiveEnsObjectIds
+    };
+    delete updatedActiveEnsObjectIds[genomeIdToRemove];
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+
+    browserStorageService.updateActiveEnsObjectIds({
+      ...currentActiveEnsObjectIds,
+      [genomeIdToRemove]: undefined
+    });
```

Comment:
We also need to clean all this: https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/browser/browser-storage-service.ts#L69-L89

Reply:
That's done